src/components/Navbar/SideWidgets/CurrentSpeed.vue

```diff
@@ -1,12 +1,40 @@
 <script setup lang="ts">
 import SpeedCard from '@/components/Core/SpeedCard.vue'
+import SpeedCardClickable from '@/components/Core/SpeedCardClickable.vue'
 import { useMaindataStore, useVueTorrentStore } from '@/stores'
 import { storeToRefs } from 'pinia'
 import { useI18nUtils } from '@/composables'
+import { useTorrentStore } from '@/stores'
+import { TorrentState } from '@/constants/vuetorrent'
+import { computed } from 'vue'
 
 const { t } = useI18nUtils()
 const { serverState } = storeToRefs(useMaindataStore())
 const { displayGraphLimits } = storeToRefs(useVueTorrentStore())
+const { statusFilter } = storeToRefs(useTorrentStore())
+
+const downloadingStates = [TorrentState.DOWNLOADING, TorrentState.DL_FORCED, TorrentState.FORCED_META_DOWNLOAD]
+const uploadingStates = [TorrentState.UPLOADING, TorrentState.UL_FORCED]
+
+function removeFilter() {
+  statusFilter.value = []
+}
+
+function selectDownloading() {
+  statusFilter.value = downloadingStates
+}
+
+function selectUploading() {
+  statusFilter.value = uploadingStates
+}
+
+const isDownloadingFilterActive = computed(
+  () => downloadingStates.every(state => statusFilter.value.includes(state)) && downloadingStates.length == statusFilter.value.length
+)
+
+const isUploadingFilterActive = computed(
+  () => statusFilter.value.every(state => uploadingStates.includes(state)) && uploadingStates.length == statusFilter.value.length
+)
```

Comment:
Use type assertion as well

```suggestion
const isDownloadingFilterActive = computed(
  () => downloadingStates.every(state => statusFilter.value.includes(state)) && downloadingStates.length === statusFilter.value.length
)

const isUploadingFilterActive = computed(
  () => statusFilter.value.every(state => uploadingStates.includes(state)) && uploadingStates.length === statusFilter.value.length
)
```